src/components/RulesModal.tsx

```
@@ -0,0 +1,174 @@
+  const elements: React.ReactNode[] = [];
+  let currentList: string[] = [];
+  let listType: 'ul' | 'ol' | null = null;
+  let keyCounter = 0;
+
+  const flushList = () => {
+    if (currentList.length > 0 && listType) {
+      const ListTag = listType;
+      elements.push(
+        <ListTag key={`list-${keyCounter++}`}>
+          {currentList.map((item, i) => (
+            <li key={i} dangerouslySetInnerHTML={{ __html: formatInline(item) }} />
```

Biome (2.4.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

(lint/security/noDangerouslySetInnerHtml)